src/client.rs

```diff
@@ -295,32 +311,57 @@ where
             Err(_) => return Err(Error::SocketError),
         };
 
-        // Compare server and client hashes.
-        let password_hash = md5_hash_password(&config.user.name, &config.user.password, &salt);
 
-        if password_hash != password_response {
-            debug!("Password authentication failed");
-            wrong_password(&mut write, &config.user.name).await?;
-            return Err(Error::ClientError);
+
+        let mut target_pool: ConnectionPool = ConnectionPool::default();
+        let mut transaction_mode = false;
+
+        if admin {
+            let correct_user = config.general.admin_username.as_str();
+            let correct_password = config.general.admin_password.as_str();
+
+            // Compare server and client hashes.
+            let password_hash = md5_hash_password(correct_user, correct_password, &salt);
+            if password_hash != password_response {
+                debug!("Password authentication failed");
+                wrong_password(&mut write, user).await?;
+                return Err(Error::ClientError);
+            }
+        } else {
+            target_pool = match get_pool(database.clone(), user.clone()) {
+                Some(pool) => pool,
+                None => {
+                    error_response(
+                        &mut write,
+                        &format!("No pool configured for database: {:?}, user: {:?}", database, user),
+                    ).await?;
+                    return Err(Error::ClientError)
+                }
+            };
+            transaction_mode = target_pool.settings.pool_mode == "transaction";
+
+            // Compare server and client hashes.
+            let correct_password = target_pool.settings.user.password.as_str();
+            let password_hash = md5_hash_password(user, correct_password, &salt);
+
+            if password_hash != password_response {
+                debug!("Password authentication failed");
+                wrong_password(&mut write, user).await?;
+                return Err(Error::ClientError);
+            }
         }
 
         debug!("Password authentication successful");
 
         auth_ok(&mut write).await?;
-        write_all(&mut write, get_pool().server_info()).await?;
-        backend_key_data(&mut write, process_id, secret_key).await?;
+        if !admin {
```

Comment:
You can send this for admin users too, it won't hurt I think.

Reply:
For admin, there isn't really any server, and the pool we use to get `server_info` is a dummy one.

Reply:
Yeah, but if you don't send server info, the psql will look weird. We prewarm the pool on pooler boot, so we actually have server info, so we have something to send to the client.